sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -170,42 +193,116 @@ private void validateDocCRUDandQuery() throws Exception {
-                assertThat(e.getMessage().contains("Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.\n"));
-            }
+
+        //Document Create - Negative tests
+        //Using incomplete partition key in method params
+        PartitionKey partitionKey =
+            new PartitionKeyBuilder()
+                .add("Redmond")
+                .build();
+        try {
+            createdMultiHashContainer.createItem(doc, partitionKey, new CosmosItemRequestOptions());
+        } catch (Exception e) {
+            assertThat(e.getMessage().contains("Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.\n"));
+        }
+
+        //Using incomplete partition key in item body
+        ObjectNode wrongDoc = new ObjectNode(JSON_NODE_FACTORY_INSTANCE);
+        wrongDoc.set("id", new TextNode(UUID.randomUUID().toString()));
+        wrongDoc.set("city", new TextNode("Redmond"));
+        try {
+            createdMultiHashContainer.createItem(wrongDoc);
+        } catch (Exception e) {
+            assertThat(e.getMessage().contains("PartitionKey extracted from document doesn't match the one specified in the header."));
         }
 
         //Document Read
-        {
-            for (int i = 0; i < docs.size(); i++) {
-                ObjectNode doc_current = docs.get(i);
-                PartitionKey partitionKey = new PartitionKeyBuilder()
-                    .add(doc_current.get("city").asText())
-                    .add(doc_current.get("zipcode").asText())
-                    .build();
-                CosmosItemResponse<ObjectNode> response = createdMultiHashContainer.readItem(doc_current.get("id").asText(), partitionKey, ObjectNode.class);
-            }
+        for (int i = 0; i < docs.size(); i++) {
+            ObjectNode doc_current = docs.get(i);
+            partitionKey = new PartitionKeyBuilder()
+                .add(doc_current.get("city").asText())
+                .add(doc_current.get("zipcode").asText())
+                .build();
+            CosmosItemResponse<ObjectNode> response = createdMultiHashContainer.readItem(doc_current.get("id").asText(), partitionKey, ObjectNode.class);
+            validateItemResponse(doc_current, response);
         }
 
-        // Query Tests.
+        //Document Read - Negative tests
+        //Using incomplete partition key
+        PartitionKey partialPK = new PartitionKeyBuilder().add(doc.get("city").asText()).build();
+        try {
+            createdMultiHashContainer.readItem(doc.get("id").asText(), partialPK, ObjectNode.class);
+        } catch (Exception e) {
+            assertThat(e.getMessage().contains("Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.\n"));
```

Comment:
It might be a good idea to assert that the method throws a specific type of error rather than on the message. If I remember correctly that message is sent from the backend, and it might change before GA.